examples/open_deep_research/run.py

```diff
@@ -0,0 +1,293 @@
+Give it all you can: I know for a fact that you have access to all the relevant tools to solve it and find the correct answer (the answer does exist). Failure or 'I cannot answer' or 'None found' will not be tolerated, success will be rewarded.
+Run verification steps if that's needed, you must make sure you find the correct answer!
+Here is the task:
+""" + example["question"]
+
+    if example["file_name"]:
+        if ".zip" in example["file_name"]:
+            prompt_use_files = "\n\nTo solve the task above, you will have to use these attached files:\n"
+            prompt_use_files += get_zip_description(
+                example["file_name"], example["question"], visual_inspection_tool, document_inspection_tool
+            )
+        else:
+            prompt_use_files = "\n\nTo solve the task above, you will have to use this attached file:"
+            prompt_use_files += get_single_file_description(
+                example["file_name"], example["question"], visual_inspection_tool, document_inspection_tool
+            )
+        augmented_question += prompt_use_files
+
+    start_time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+    try:
+        # Run agent 🚀
+        final_result = agent.run(augmented_question)
+
+        agent_memory = agent.write_memory_to_messages(summary_mode=True)
+
+        final_result = prepare_response(augmented_question, agent_memory, reformulation_model=model)
+
+        output = str(final_result)
+        for memory_step in agent.memory.steps:
+            memory_step.model_input_messages = None
+        intermediate_steps = [str(step) for step in agent.memory.steps]
+
+        # Check for parsing errors which indicate the LLM failed to follow the required format
+        parsing_error = True if any(["AgentParsingError" in step for step in intermediate_steps]) else False
+
+        # check if iteration limit exceeded
+        iteration_limit_exceeded = True if "Agent stopped due to iteration limit or time limit." in output else False
+        raised_exception = False
+
+    except Exception as e:
+        print("Error on ", augmented_question, e)
+        output = None
+        intermediate_steps = []
+        parsing_error = False
+        iteration_limit_exceeded = False
+        exception = e
+        raised_exception = True
+    end_time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+    annotated_example = {
+        "agent_name": model.model_id,
+        "question": example["question"],
+        "augmented_question": augmented_question,
+        "prediction": output,
+        "intermediate_steps": intermediate_steps,
+        "parsing_error": parsing_error,
+        "iteration_limit_exceeded": iteration_limit_exceeded,
+        "agent_error": str(exception) if raised_exception else None,
+        "start_time": start_time,
+        "end_time": end_time,
+        "task": example["task"],
```

Comment:
What about adding the task_id for easily finding a specific sample?
```suggestion
        "task": example["task"],
        "task_id": example["task_id"],
```